### PR TITLE
Support repo-based makecatalogs

### PR DESCRIPTION
**Pull request by tboyko**
This PR allows a Munki repo plugin to define its own makecatalogs behavior via a `makecatalogs()` method. If not defined, Munki will use the default behavior of `makecatalogslib.makecatalogs()`, as before.

### CODE REVIEW

File: code/client/makecatalogs [hunk 1]
```diff
@@ -95,7 +95,10 @@ def main():
         exit(-1)
 
     # Make the catalogs
-    errors = makecatalogslib.makecatalogs(repo, options, output_fn=print)
+    try:
+        errors = repo.makecatalogs()
```
Comment (gregneagle) [inline code/client/makecatalogs:99]: Do we want to pass along the same parameters to `repo.makecatalogs()`? The implementation would be free to ignore them, but perhaps they might be useful.

Comment (tboyko) [reply code/client/makecatalogs]: Good point. For my uses, they aren't needed, but make sense for some future proofing. Updating...

**Comment by tboyko** [comment]
Thanks Greg 👍

**Comment by gregneagle** [comment]
I reverted this commit in branch main, then cherry-picked it into branch Munki5dev. It will be in the next Munki 5.1 beta release.

**Pull request merged**